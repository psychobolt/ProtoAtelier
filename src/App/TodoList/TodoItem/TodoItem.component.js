// @flow
import React from 'react';
import styled from 'styled-components';

import * as styles from './TodoItem.style';

type Props = {
  onClick: () => void,
  completed: boolean,
  text: string,
  className: string
}

const TodoItem = ({ onClick, completed, text, className }: Props) => (
  <x-box class={className} completed={completed || null}>
    <x-checkbox onClick={onClick} toggled={completed || null} />
    <x-label for="checkbox">{text}</x-label>
  </x-box>
);

export default styled(TodoItem)`${styles.box}`;
