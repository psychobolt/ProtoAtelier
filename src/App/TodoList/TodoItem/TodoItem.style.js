import { css } from 'styled-components';

export const box = css`
  text-decoration: ${props => (props.completed ? 'line-through' : 'none')};
`;
