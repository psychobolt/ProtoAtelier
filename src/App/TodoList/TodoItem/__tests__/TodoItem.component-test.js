import React from 'react';
import { shallow } from 'enzyme';

import TodoItem from '../TodoItem.component';

describe('components <TodoItem />', () => {
  it('TodoItem should render correctly -- completed', () => {
    const props = {
      onClick: jest.fn(),
      completed: true,
      text: 'Item',
    };
    const wrapper = shallow(<TodoItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('TodoItem should render correctly -- not completed', () => {
    const props = {
      onClick: jest.fn(),
      completed: false,
      text: 'Item',
    };
    const wrapper = shallow(<TodoItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
