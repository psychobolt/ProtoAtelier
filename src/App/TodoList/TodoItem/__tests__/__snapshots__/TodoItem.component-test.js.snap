// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <TodoItem /> TodoItem should render correctly -- completed 1`] = `
<x-box
  style={
    Object {
      "textDecoration": "line-through",
    }
  }
>
  <x-checkbox
    onClick={[MockFunction]}
    toggled={true}
  />
  <x-label
    for="checkbox"
  >
    Item
  </x-label>
</x-box>
`;

exports[`components <TodoItem /> TodoItem should render correctly -- not completed 1`] = `
<x-box
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  <x-checkbox
    onClick={[MockFunction]}
    toggled={null}
  />
  <x-label
    for="checkbox"
  >
    Item
  </x-label>
</x-box>
`;
